src/io/webcam.rs

```diff
@@ -0,0 +1,247 @@
+        }
+    }
+
+    /// Sets the camera id for the WebcamCaptureBuilder.
+    ///
+    /// # Arguments
+    ///
+    /// * `camera_id` - The desired camera id
+    pub fn camera_id(mut self, camera_id: usize) -> Self {
+        self.camera_id = camera_id;
+        self
+    }
+
+    /// Sets the image size for the WebcamCaptureBuilder.
+    ///
+    /// # Arguments
+    ///
+    /// * `size` - The desired image size
+    pub fn with_size(mut self, size: ImageSize) -> Self {
+        self.size = Some(size);
+        self
+    }
+
+    /// Create a new [`WebcamCapture`] object.
+    pub fn build(self) -> Result<WebcamCapture> {
+        WebcamCapture::new(self.camera_id, self.size)
+    }
+}
+
+impl Default for WebcamCaptureBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// A webcam capture object that grabs frames from the camera
+/// using GStreamer.
+///
+/// # Example
+///
+/// ```no_run
+/// use kornia_rs::{image::ImageSize, io::webcam::WebcamCaptureBuilder};
+///
+/// fn main() -> Result<(), Box<dyn std::error::Error>> {
+///   // create a webcam capture object with camera id 0
+///   // and force the image size to 640x480
+///   let webcam = WebcamCaptureBuilder::new()
+///     .camera_id(0)
+///     .with_size(ImageSize {
+///       width: 640,
+///       height: 480,
+///   })
+///   .build()?;
+///
+///   // start grabbing frames from the camera
+///   webcam.run(|img| {
+///     println!("Image: {:?}", img.size());
+///   })?;
+///
+///   Ok(())
+/// }
+/// ```
+pub struct WebcamCapture {
+    pipeline: gst::Pipeline,
+    receiver: std::sync::mpsc::Receiver<Image<u8, 3>>,
+    handle: Option<std::thread::JoinHandle<()>>,
+}
+
+impl WebcamCapture {
+    /// Creates a new WebcamCapture object.
+    ///
+    /// # Arguments
+    ///
+    /// * `camera_id` - The camera id used for capturing images
+    /// * `size` - The image size used for resizing directly from the camera
+    ///
+    /// # Returns
+    ///
+    /// A WebcamCapture object
+    fn new(camera_id: usize, size: Option<ImageSize>) -> Result<Self> {
+        gst::init()?;
+
+        // create a pipeline specified by the camera id and size
+        let pipeline_str = Self::gst_pipeline_string(camera_id, size);
+        let pipeline = gst::parse::launch(&pipeline_str)?
+            .downcast::<gst::Pipeline>()
+            .map_err(|_| anyhow::anyhow!("Failed to downcast pipeline"))?;
+
+        let appsink = pipeline
+            .by_name("sink")
+            .ok_or_else(|| anyhow::anyhow!("Failed to get sink"))?
+            .dynamic_cast::<gst_app::AppSink>()
+            .map_err(|_| anyhow::anyhow!("Failed to cast to AppSink"))?;
+
+        let (tx, rx) = std::sync::mpsc::channel();
+
+        appsink.set_callbacks(
+            gst_app::AppSinkCallbacks::builder()
+                .new_sample(move |sink| match Self::extract_image_frame(sink) {
+                    Ok(frame) => {
+                        if let Err(e) = tx.send(frame) {
```

Comment:
if one doesn't start processing messages via `run`, would this keep on populating the channel until the application runs out of memory?